src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -748,6 +748,18 @@ def __call__(self):
         cmsswStep = self.step.getTypeHelper()
         for om in cmsswStep.listOutputModules():
             mod = cmsswStep.getOutputModule(om)
+            modName = str(getattr(mod, "_internal_name"))
+
+            if funcName == 'merge':
+                # Do not use both Merged output label unless useErrorDataset is False
+                # Do not use both MergedError output label unless useErrorDataset is True 
+                useErrorDataset = getattr(self.jobBag, "useErrorDataset", False)
+
+                if modName.startswith('Merged'):
```

Comment:
If the module name is either `Merged` or `MergedError`, then I would suggest to make a full comparison, something like:
```
if useErrorDataset and modName != 'MergedError':
    continue
if not useErrorDataset and modName == 'MergedError':
    continue
```

The `if` statement in the line above could go away as well.